huaweicloud/services/waf/data_source_huaweicloud_waf_certificate.go

```diff
@@ -0,0 +1,88 @@
+package waf
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/huaweicloud/golangsdk/openstack/waf/v1/certificates"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+)
+
+const (
+	// EXP_STATUS_NOT_EXPIRED not expired
+	EXP_STATUS_NOT_EXPIRED = 0
+	// EXP_STATUS_EXPIRED has expired
+	EXP_STATUS_EXPIRED = 1
+	// EXP_STATUS_EXPIRED_SOON will expire soon
+	EXP_STATUS_EXPIRED_SOON = 2
+
+	DEFAULT_PAGE_NUM  = 1
+	DEFAULT_PAGE_SIZE = 5
+)
+
+func DataSourceWafCertificateV1() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceWafCertificateV1Read,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"expire_status": {
```

Comment:
to keep with the resource, suggest to add `expiration` attribute.

Reply:
added